resources/functionMap.php

```diff
@@ -4705,167 +4705,167 @@
 'Imagick::getImageFormat' => ['string'],
 'Imagick::getImageGamma' => ['float'],
-'Imagick::getImageGeometry' => ['array'],
-'Imagick::getImageGravity' => ['int'],
-'Imagick::getImageGreenPrimary' => ['array'],
+'Imagick::getImageGeometry' => ['array{width:int,height:int}'],
+'Imagick::getImageGravity' => ['Imagick::GRAVITY_*'],
+'Imagick::getImageGreenPrimary' => ['array{x:float,y:float}'],
 'Imagick::getImageHeight' => ['int'],
 'Imagick::getImageHistogram' => ['list<ImagickPixel>'],
 'Imagick::getImageIndex' => ['int'],
-'Imagick::getImageInterlaceScheme' => ['int'],
-'Imagick::getImageInterpolateMethod' => ['int'],
+'Imagick::getImageInterlaceScheme' => ['Imagick::INTERLACE_*'],
+'Imagick::getImageInterpolateMethod' => ['Imagick::INTERPOLATE_*'],
 'Imagick::getImageIterations' => ['int'],
-'Imagick::getImageLength' => ['int'],
+'Imagick::getImageLength' => ['0|positive-int'],
 'Imagick::getImageMagickLicense' => ['string'],
 'Imagick::getImageMatte' => ['bool'],
 'Imagick::getImageMatteColor' => ['ImagickPixel'],
-'Imagick::getImageMimeType' => ['string'],
-'Imagick::getImageOrientation' => ['int'],
-'Imagick::getImagePage' => ['array'],
+'Imagick::getImageMimeType' => ['non-empty-string'],
+'Imagick::getImageOrientation' => ['Imagick::ORIENTATION_*'],
+'Imagick::getImagePage' => ['array{width:int,height:int,x:int,y:int}'],
 'Imagick::getImagePixelColor' => ['ImagickPixel', 'x'=>'int', 'y'=>'int'],
 'Imagick::getImageProfile' => ['string', 'name'=>'string'],
 'Imagick::getImageProfiles' => ['array', 'pattern='=>'string', 'only_names='=>'bool'],
 'Imagick::getImageProperties' => ['array', 'pattern='=>'string', 'only_names='=>'bool'],
 'Imagick::getImageProperty' => ['string', 'name'=>'string'],
-'Imagick::getImageRedPrimary' => ['array'],
+'Imagick::getImageRedPrimary' => ['array{x:float,y:float}'],
 'Imagick::getImageRegion' => ['Imagick', 'width'=>'int', 'height'=>'int', 'x'=>'int', 'y'=>'int'],
-'Imagick::getImageRenderingIntent' => ['int'],
-'Imagick::getImageResolution' => ['array'],
+'Imagick::getImageRenderingIntent' => ['Imagick::RENDERINGINTENT_*'],
+'Imagick::getImageResolution' => ['array{x:float,y:float}'],
 'Imagick::getImagesBlob' => ['string'],
-'Imagick::getImageScene' => ['int'],
+'Imagick::getImageScene' => ['0|positive-int'],
 'Imagick::getImageSignature' => ['string'],
-'Imagick::getImageSize' => ['int'],
-'Imagick::getImageTicksPerSecond' => ['int'],
+'Imagick::getImageSize' => ['0|positive-int'],
+'Imagick::getImageTicksPerSecond' => ['0|positive-int'],
 'Imagick::getImageTotalInkDensity' => ['float'],
-'Imagick::getImageType' => ['int'],
+'Imagick::getImageType' => ['Imagick::IMGTYPE_*'],
 'Imagick::getImageUnits' => ['int'],
 'Imagick::getImageVirtualPixelMethod' => ['int'],
-'Imagick::getImageWhitePoint' => ['array'],
-'Imagick::getImageWidth' => ['int'],
-'Imagick::getInterlaceScheme' => ['int'],
+'Imagick::getImageWhitePoint' => ['array{x:float,y:float}'],
+'Imagick::getImageWidth' => ['0|positive-int'],
```

Comment:
Oddly, [`getImageWidth`](https://github.com/Imagick/imagick/blob/28f27044e435a2b203e32675e942eb8de620ee58/imagick_class.c#L8010) is declared as `unsigned long width`, but [`getImageHeight`](https://github.com/Imagick/imagick/blob/28f27044e435a2b203e32675e942eb8de620ee58/imagick_class.c#L7434) seems to be just `long height`.